File: lua/bob/bob.lua
Bob = {}

function Bob.hey (input_str)
    if input_str == '' then
        return 'Fine, be that way.'
    end
    if input_str:match('?$') then
        return 'Sure'
    end
    _, count = input_str:gsub('[A-Z!]', '')
    if count > 2 then
        return 'Whoa, chill out!'
    end
    return 'Whatever'
end

return Bob

File: scala/bob/src/main/scala/bob.scala
class Bob {
    def hey(statement:String) : String = {
        if (statement.exists(_.isLetter) && statement.toUpperCase == statement) "Whoa, chill out!"
        else if (statement.endsWith("?")) "Sure."
        else if (statement.trim.isEmpty) "Fine. Be that way!"
        else "Whatever."
    }
}